packages/custom-objects-exporter/src/cli.js

```diff
@@ -0,0 +1,115 @@
+import fs from 'fs'
+import { getCredentials } from '@commercetools/get-credentials'
+import npmlog from 'npmlog'
+import PrettyError from 'pretty-error'
+import yargs from 'yargs'
+
+import CustomObjectsExporter from './main'
+import { description } from '../package.json'
+
+process.title = 'custom-objects-exporter'
+
+const args = yargs
+  .usage(
+    `
+Usage: $0 [options]
+${description}`
+  )
+  .showHelpOnFail(false)
+  .help('help', 'Show help text.')
+  .version()
+  .option('output', {
+    alias: 'o',
+    default: 'stdout',
+    describe: 'Path to output file.',
+  })
+  .coerce('output', arg => {
+    if (arg !== 'stdout') return fs.createWriteStream(String(arg))
+
+    return process.stdout
+  })
+  .option('apiUrl', {
+    default: 'https://api.sphere.io',
+    describe: 'The host URL of the HTTP API service.',
+  })
+  .option('authUrl', {
+    default: 'https://auth.sphere.io',
+    describe: 'The host URL of the OAuth API service.',
+  })
+  .option('accessToken', {
+    describe: 'CTP client access token.',
+  })
+  .option('projectKey', {
+    alias: 'p',
+    describe: 'API project key.',
+    demand: true,
+  })
+  .option('where', {
+    alias: 'w',
+    describe: 'Specify where predicate.',
+  })
+  .option('logLevel', {
+    default: 'info',
+    describe: 'Logging level: error, warn, info or verbose.',
+  })
+  .option('logFile', {
+    default: 'custom-objects-export.log',
+    describe: 'Path to file where to save logs.',
+  })
+  .coerce('logLevel', arg => {
+    npmlog.level = arg
+  }).argv
+
+const logError = error => {
+  const errorFormatter = new PrettyError()
+
+  if (npmlog.level === 'verbose')
+    process.stderr.write(`ERR: ${errorFormatter.render(error)}`)
+  else process.stderr.write(`ERR: ${error.message || error}`)
+}
+
+// print errors to stderr if we use stdout for data output
+// if we save data to output file errors are already logged by npmlog
+const errorHandler = errors => {
+  if (Array.isArray(errors)) errors.forEach(logError)
+  else logError(errors)
+
+  process.exitCode = 1
+}
+
+const resolveCredentials = _args => {
+  if (_args.accessToken) return Promise.resolve({})
+  return getCredentials(_args.projectKey)
+}
+
+// If the stdout is used for a data output, save all logs to a log file.
+if (args.output === process.stdout)
+  npmlog.stream = fs.createWriteStream(args.logFile)
+else npmlog.stream = process.stdout
+
+// Register error listener
+args.output.on('error', errorHandler)
+
+resolveCredentials(args)
+  .then(credentials => {
+    const apiConfig = {
+      host: args.authUrl,
+      apiUrl: args.apiUrl,
+      projectKey: args.projectKey,
+      credentials,
+    }
+    const constructorOptions = {
```

Comment:
Hm, isn't the logger also a contructor option? Should we name this according to where it ends up?

Reply:
Like `exporterOptions`?